src/kudu/consensus/consensus_queue.cc

```diff
@@ -779,7 +779,12 @@ Status PeerMessageQueue::RequestForPeer(const string& uuid,
   // If the next hop != the destination, we are sending these messages via a proxy.
   bool route_via_proxy = *next_hop_uuid != uuid;
   if (route_via_proxy) {
+    // Set proxy uuid
     request->set_proxy_dest_uuid(*next_hop_uuid);
+  } else {
+    // Clear proxy uuid to ensure that this message is not rejected by the
+    // destination
+    request->clear_proxy_dest_uuid();
```

Comment:
Wouldn't it be better to do this upfront before we call RequestForPeer in ::SendNextRequest.
A generic request reinitialization function.

Reply:
Yes, seems like a good idea. But do not want to do it in this diff. Not sure if there is anything that relies on (intentionally or unintentionally) some fields in the request never being cleared. Is that okay?